src/metadata-parser.h

```diff
@@ -70,20 +70,33 @@ namespace librealsense
         bool try_get(const frame& frm, rs2_metadata_type& result) const
         {
             const uint8_t* pos = frm.additional_data.metadata_blob.data();
-            while (pos <= frm.additional_data.metadata_blob.data() + frm.additional_data.metadata_blob.size())
+            while( pos <= frm.additional_data.metadata_blob.data() + frm.additional_data.metadata_blob.size() )
             {
-                const rs2_frame_metadata_value* type = reinterpret_cast<const rs2_frame_metadata_value*>(pos);
-                pos += sizeof(rs2_frame_metadata_value);
-                if (_type == *type)
+                const rs2_frame_metadata_value* type = reinterpret_cast< const rs2_frame_metadata_value* >( pos );
+                pos += sizeof( rs2_frame_metadata_value );
+                if( _type == *type )
                 {
-                    const rs2_metadata_type* value = reinterpret_cast<const rs2_metadata_type*>(pos);
-                    memcpy((void*)&result, (const void*)value, sizeof(*value));
+                    const rs2_metadata_type* value = reinterpret_cast< const rs2_metadata_type* >( pos );
+                    memcpy( (void*)&result, (const void*)value, sizeof( *value ) );
                     return true;
                 }
-                pos += sizeof(rs2_metadata_type);
+                pos += sizeof( rs2_metadata_type );
             }
+
+            //Search extra_data for software device, might be used if metadata_blob is not big enough for all metadata
+            if( !frm.additional_data.sw_device_extra_data.empty() )
```

Comment:
No way to forward the metadata lookup to the frame, so that we could simply provide extra checks in a software frame?

Reply:
We can use some other, new kind of metadata_parser for dds frames but it was less efficient (when tested with the viewer, which reads all metadata fields)